File: javascript/leetcode/1213. Intersection of Three Sorted Arrays.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @param {number[]} arr3
 * @return {number[]}
 */
var arraysIntersection = function(arr1, arr2, arr3) {
    let ans = new Array()
    
    const first = new Set(arr1)
    const second = new Set(arr2)
    const third = new Set(arr3)
    
    for(const i of arr1) 
        if(first.has(i) && second.has(i) && third.has(i))
            ans.push(i)
    
    ans.sort((a, b) => a - b)
    
    return ans
}; 

File: c++/leetcode/1213. Intersection of Three Sorted Arrays.cpp
class Solution {
public:
    vector<int> arraysIntersection(vector<int>& arr1, vector<int>& arr2, vector<int>& arr3) {
        vector<int> ans;
        unordered_set<int> first(arr1.begin(), arr1.end());
        unordered_set<int> second(arr2.begin(), arr2.end());
        unordered_set<int> third(arr3.begin(), arr3.end());
        
        for(const auto& i : arr1) 
            if(first.find(i) != first.end() && second.find(i) != second.end() &&
              third.find(i) != third.end())
                ans.push_back(i);
        
        sort(ans.begin(), ans.end());      
        
        return ans;
    }
};
